pkg/lightning/backend/external/byte_reader.go

```diff
@@ -312,27 +313,33 @@ func (r *byteReader) reload() error {
 		r.curBufOffset = 0
 		return nil
 	}
-	// when not using concurrentReader, len(curBuf) == 1
-	n, err := io.ReadFull(r.storageReader, r.curBuf[0][0:])
-	if err != nil {
-		switch {
-		case goerrors.Is(err, io.EOF):
-			// move curBufIdx so following read will also find EOF
-			r.curBufIdx = len(r.curBuf)
-			return err
-		case goerrors.Is(err, io.ErrUnexpectedEOF):
-			// The last batch.
-			r.curBuf[0] = r.curBuf[0][:n]
-		case goerrors.Is(err, context.Canceled):
-			return err
-		default:
-			r.logger.Warn("other error during read", zap.Error(err))
-			return err
+
+	return util.RunWithRetry(util.DefaultMaxRetries, util.RetryInterval, func() (retryable bool, err error) {
+		// when not using concurrentReader, len(curBuf) == 1
+		n, err := io.ReadFull(r.storageReader, r.curBuf[0])
+		if err != nil {
+			switch {
+			case goerrors.Is(err, io.EOF):
+				// move curBufIdx so following read will also find EOF
+				r.curBufIdx = len(r.curBuf)
+				return false, err
+			case goerrors.Is(err, io.ErrUnexpectedEOF):
+				if n == 0 {
+					return true, err
```

Comment:
maybe print log for this unexpected case. For `(0, ErrUnexpectedEOF)` lasts for a long time and still fails after retry, we can see it in log.